packages/dapp/src/components/RenewTLDModal.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="action-modal">
+    <v-modal ref="modal">
+      <template v-slot:header>
+        <h3>You're almost done </h3>
+      </template>
+
+      <template v-slot:body>
+        <div class="modal-subtitle">
+          <p>To finish the transaction, please check the informations above to ensure all itens are correct and confirm the transaction in Metamask</p>
+        </div>
+        <div class="modal-content">
+          <div class="modal-row">
+            <h4>TLD:</h4>
+            <div class="row-wrapper">
+              <img class="row-content row-content--icon" src="../assets/images/check.png" width="21px" height="21px">
+              <span class="row-content row-content--data">{{tld}}</span>
+            </div>
+          </div>
+        </div>
+      </template>
+      <template v-slot:footer>
+        <button class="btn btn--success btn-confirm" @click="handleConfirm">Confirm Transaction</button>
+      </template>
+    </v-modal>
+  </div>
+</template>
+
+<script>
+import BaseModal from '@/components/BaseModal'
+
+export default {
+  name: 'RenewTLDModal',
+  extends: BaseModal,
+  components: {
+    'v-modal': BaseModal
+  },
+  props: {
+    tld: String
+  },
+  methods: {
+    openModal: function () {
+      this.$refs.modal.openModal()
+    },
+    handleConfirm: function () {
+      this.renewTopLevelDomain(this.tld)
+    },
+    renewTopLevelDomain (topLevelDomain) {
+      console.log(this.$store.getters.jnsInstance())
+
+      this.loader = this.$loading.show({
+        container: this.fullPage ? null : this.$refs.formContainer
+      })
```

Comment:
There's a problem with this approach:

Whenever you call this method, you are re-assigning the `loader` prop of the component. If two or more calls are made to a method with the same behavior, we risk having the reference overwritten by the other method, which could lead to dangling loader overlays and memory leaks.

My suggestion is to simply declare a local variable:
```js
const loader = this.$loading.show(...)
```
And then, within the `finally` block, simply do:

```js
loader.hide()
```
This way we would have no problems with dangling loaders or memory leaks.